modules/farm_fd2/src/entrypoints/transplanting/lib.submit.unit.cy.js

```diff
@@ -0,0 +1,221 @@
+import { lib } from './lib.js';
+import { lib as traySeedingLib } from '../tray_seeding/lib.js';
+import * as farmosUtil from '@libs/farmosUtil/farmosUtil';
+
+describe('Submission using the transplanting lib.', () => {
+  let traySeedingForm = {
+    seedingDate: '1950-01-02',
+    cropName: 'BROCCOLI',
+    locationName: 'CHUAU',
+    trays: 25,
+    traySize: '200',
+    seedsPerCell: 3,
+    comment: 'A comment',
+  };
+
+  let form = {
+    cropName: 'BROCCOLI',
+    picked: [],
+    transplantingDate: '1950-01-02',
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    bedFeet: 100,
+    bedWidth: 60,
+    rowsPerBed: '1',
+    equipment: ['Tractor'],
+    depth: 6,
+    speed: 5,
+    comment: 'A comment',
+  };
+
+  let cropToTerm = null;
+  let result = null;
+  let fieldMap = null;
+  let bedMap = null;
+
+  before(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+
+    cy.wrap(farmosUtil.getFieldIdToAssetMap()).then((map) => {
+      fieldMap = map;
+    });
+
+    cy.wrap(farmosUtil.getBedIdToAssetMap()).then((map) => {
+      bedMap = map;
+    });
+
+    cy.wrap(farmosUtil.getCropNameToTermMap()).then((map) => {
+      cropToTerm = map;
+    });
+
+    cy.wrap(traySeedingLib.submitForm(traySeedingForm), { timeout: 10000 })
+      .then(() => {
+        return cy.wrap(farmosUtil.getSeedlings(traySeedingForm.cropName), {
+          timeout: 10000,
+        });
+      })
+      .then((res) => {
+        form.picked[0] = { trays: 25, data: res[res.length - 1] };
+        return cy.wrap(lib.submitForm(form), { timeout: 10000 });
+      })
+      .then((res) => {
+        result = res;
+      });
+  });
+
+  beforeEach(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+  });
+
+  afterEach(() => {
+    cy.saveLocalStorage();
+    cy.saveSessionStorage();
+  });
+
+  it('Check the parents', () => {
+    expect(result.parents[0].id).to.equal(form.picked[0].data.asset_uuid);
+    expect(result.parents[0].type).to.equal('asset--plant');
+    expect(result.parents[0].attributes.name).to.equal(
+      form.picked[0].data.date + '_' + form.picked[0].data.crop
+    );
+  });
+
+  it('Check the tray inventory quantity--standard assets', () => {
+    expect(result.trayInventoryQuantities[0].attributes.value.decimal).to.equal(
+      form.picked[0].trays
+    );
+    expect(result.trayInventoryQuantities[0].type).to.equal(
+      'quantity--standard'
+    );
+    expect(result.trayInventoryQuantities[0].attributes.measure).to.equal(
+      'count'
+    );
+    expect(result.trayInventoryQuantities[0].attributes.label).to.equal(
+      'Trays'
+    );
+    expect(
+      result.trayInventoryQuantities[0].attributes.inventory_adjustment
+    ).to.equal('decrement');
+  });
```

Comment:
We should check the `asset` here to so that we are sure that the inventory of the correct plant asset is being adjusted.  This can be checked against `results.plantAsset.id`